paint/data/stac_client.py

```diff
@@ -0,0 +1,799 @@
+                        mappings.ELEVATION: item.geometry.get("coordinates")[2],
+                        mappings.DATETIME: item.datetime,
+                        "item_id": item.id,
+                    }
+                )
+            else:
+                raise ValueError(
+                    f"The collection {heliostat_collection.id.split('-')[0]} is not valid"
+                )
+        pbar.update(1)
+
+    def get_heliostat_metadata(
+        self,
+        heliostats: Union[list[str], None] = None,
+        collections: Union[list[str], None] = None,
+    ) -> None:
+        """
+        Download metadata for desired heliostats.
+
+        Parameters
+        ----------
+        heliostats : list[str], optional
+            Heliostats for which the metadata should be downloaded. If no list is provided, the metadata for all
+            heliostats is downloaded (Default is `None`).
+        collections: list[str], optional
+            Collection for which metadata should be downloaded. If now list is provided, the metadata for all
+            collections is downloaded (Default is `None`).
+        """
+        # Log warning if no collection keys provided.
+        if collections is None:
+            log.warning(
+                "No collections selected - downloading data for all collections!"
+            )
+            collections = [
+                mappings.SAVE_DEFLECTOMETRY.lower(),
+                mappings.SAVE_CALIBRATION.lower(),
+                mappings.SAVE_PROPERTIES.lower(),
+            ]
+        # Check if collection keys provided are acceptable.
+        else:
+            allowed_values = {
+                mappings.SAVE_DEFLECTOMETRY.lower(),
+                mappings.SAVE_CALIBRATION.lower(),
+                mappings.SAVE_PROPERTIES.lower(),
+            }
+            for item in collections:
+                if item not in allowed_values:
+                    raise ValueError(
+                        f"The heliostat collection must be one of: `deflectometry`, `calibration`, "
+                        f"`properties`! The key `{item}` is not accepted!"
+                    )
+        if heliostats is None:
+            save_description = "all_heliostats"
+            root = self.get_catalog(href=mappings.CATALOGUE_URL)
+            log.info("Loading all children in root catalog - please be patient!")
+            all_children = list(
+                root.get_children()
+            )  # Convert generator to list to save time later.
+        else:
+            # Find the catalogs for each desired heliostat.
+            log.info("Loading catalogs for desired heliostats. ")
+            all_children = [
+                self.get_catalog(
+                    href=mappings.HELIOSTAT_CATALOG_URL % (heliostat, heliostat)
+                )
+                for heliostat in heliostats
+            ]
+            save_description = f"{len(all_children)}_heliostats"
```

Comment:
The description when specifiying a number of heliostats to download via their ID is just identified by the number of considered heliostats, right? If I understand this correctly, this could lead to conflicts later on when downloading data for different sets of heliostats of the same size, e.g., `[1, 2, 3]` and `[4, 5, 6]`. Maybe we can additionally introduce a UUID or a timestamp in the filename?

Reply:
Good Idea - I have replaced the description with a timestamp :)